### DISCUSSION TITLE
Status of Avalonia for mobile

### DISCUSSION THREAD

**Discussion by florianwachs** [AvaloniaUI/Avalonia#9255] [Q&A] [answered]
Hi!
I`m a .net backend guy who wants to get into mobile development :-). I looked at MAUI since Preview 1, but quality and performance are not there yet (in my opinion for at least another year...). Took a look at flutter and really liked how it works. Still would prefer a .NET solution. Now, I saw that Avalonia 11 is coming with Android / IOS support and I saw a cool Wordle implementation. My questions (finally :-) ):
- Is there any guidance on how to debug  / deploy to a physical device?
- are there any "performance numbers" (like the LOL´s per seconds demo)
- are there libs to access hardware? Like Maui.Essentials

Basically I want to know if any of you is planning a real life mobile app.

I have to say that Avalonia looks so promising and I really hope it takes off with mobile too.

Thanks a lot and thanks for this great framwork.

**Comment by robloo** [comment]
The core maintainers will have to give you some feedback here. However, there are upcoming commercial apps intending to use the mobile backend.

You can gauge performance with the screen-captures on the associated PRs:

 * IPad : https://github.com/AvaloniaUI/Avalonia/pull/7565#issuecomment-1049496241 With M1, iPhone should be comparable though and even better than Android.
 * Android : https://github.com/AvaloniaUI/Avalonia/pull/8691#issuecomment-1207313541. This is after a major performance fix and is only getting better from here.

Because Avalonia fully renders itself using Skia -- the same as Flutter -- mobile platforms should stabilize very quickly. The vast majority of code is the same on all platforms and desktop is already ready and used in commercial apps.

**Reply by florianwachs** [reply]
@robloo thank you very much for your input!

**Comment by lukaszossowski** [comment, marked as answer]
I can't really talk about iOS but for Android here are my answers.

>     * Is there any guidance on how to debug  / deploy to a physical device?

Yes, with VS you do it just like you would with Xamarin/Xamarin.Forms. Both MSDN and Android have helpful documentation in this matter. It was fairly easy even for a beginner like me.

>     * are there any "performance numbers" (like the LOL´s per seconds demo)

No idea. It does seem to run pretty well for me, even though I have a pretty control heavy app in the works right now.

>     * are there libs to access hardware? Like Maui.Essentials

MSDN and Android again will come to the rescue. You can use Xamarin.Essentials or the Android libraries.

> Basically I want to know if any of you is planning a real life mobile app.

Apart from some issues here and there it has been an incredible experience for me. I had been working on a desktop app for a while now and looking around for porting it to Android as well. I've tried Xamarin.Forms and Maui, Uno Platform but neither of those really worked for me. But then came the 11 update for Avalonia and it pushed me to try Avalonia mobile support. A couple hours later I was reusing everything I've already written for my desktop app and had it running on my phone with all those complex controls, animations etc. I could only imagine the amount of custom work I would have to do in any other .NET ui framework, not even mentioning far fewer resources online on that matter - certainly compared to Avalonia which really benefits from almost all WPF content online.

**Reply by robloo** [reply]
This is very interesting for me to read as well. An android port to Avalonia on my end hasn't started yet. However, a port was completed in the past using the Uno Platform and essentially failed due to bugs and performance issues. This was a desktop-class app fully responsive and adaptive for a mobile form factor. However, Uno simply couldn't handle it.

Xamarin Forms, now Maui, is a non-starter due to the architecture differences and knowing there will be more bugs and performance issues going that route -- not to mention a monumental effort for custom controls. Avalonia was the clear choice a while back, just been waiting for it to catch up in mobile and I think we are there now.

I've been watching the maturing stabilize in Avalonia for Android rapidly over the last few months. Really looking forward to seeing what it can do with a moderately sized mobile app soon myself. Your comments make me look forward to the porting process.

**Comment by florianwachs** [comment]
@lukaszossowski thanks so much for the detailed information
